src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -80,7 +80,13 @@ public abstract class TestResult : ITestResult
         private string _message;
         private string _stackTrace;
 
-        private List<AssertionResult> _assertionResults = new List<AssertionResult>();
+        private readonly List<AssertionResult> _assertionResults = new List<AssertionResult>();
+
+        /// <summary>
+        /// TestAttachments attached to result.
+        /// Only exposed for testing.
+        /// </summary>
+        internal ICollection<TestAttachment> TestAttachments { get; } = new List<TestAttachment>();
```

Comment:
Ah! I didn't notice the comment. So this field is not the private backing field for a public property then? That seems a bit odd, since it leaves a part of the TestResult inaccessible to clients.